jerry-core/ecma/builtin-objects/ecma-builtin-object.c

```diff
@@ -94,6 +97,67 @@ ecma_builtin_object_dispatch_construct (const ecma_value_t *arguments_list_p, /*
   }
 } /* ecma_builtin_object_dispatch_construct */
 
+#ifndef CONFIG_DISABLE_ES2015_BUILTIN
+/**
+ * [[GetPrototypeOf]]
+ *
+ * See also:
+ *          ES2015 9.1.1
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value
+ */
+static ecma_value_t
+ecma_builtin_object_object_get_prototype_of (ecma_value_t this_arg,
+                                            ecma_value_t arg)
+{
+  JERRY_UNUSED (this_arg);
+  ecma_value_t ret_value = ECMA_VALUE_EMPTY;
+  bool was_object = ecma_is_value_object (arg);
+
+  /* 1. */
+  if (!was_object)
+  {
+    if (ecma_is_value_boolean (arg))
+    {
+      arg = ecma_op_create_boolean_object (arg);
+    }
+    if (ecma_is_value_number (arg))
+    {
+      arg = ecma_op_create_number_object (arg);
+    }
+    if (ecma_is_value_string (arg))
+    {
+      arg = ecma_op_create_string_object (&arg, 1);
+    }
+    if (ecma_is_value_null (arg) || ecma_is_value_undefined (arg))
+    {
+      ret_value = ecma_raise_type_error (ECMA_ERR_MSG ("Cannot coerce null or undefined to object"));
+    }
```

Comment:
I think there is a `ecma_op_to_object` in `ecma-conversion.h`, why don't we use it?